### PR TITLE
[ISSUE #3028] Refactor ThreadFactory implement

### LINKED ISSUE #3028: [Enhancement] Refactor ThreadFactory implement
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

There are several implementations of ThreadFactory in the project now, and we need to refactor them to be uniform.The patterns for creating ThreadFactory are diverse and need to be unified.
image
image

### Describe the solution you'd like

Implement a custom ThreadFactory and place the implementation in a common module. A utility class to create thread pools can be provided, but it's not mandatory.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by mxsm**
Fixes #3028 

### Motivation

*Explain the content here.*
*Explain why you want to make the changes and what problem you're trying to solve.*

### Modifications

- *Refactor ThreadFactory implement.*

### Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
- If a feature is not applicable for documentation, explain why?
- If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: eventmesh-common/src/main/java/org/apache/eventmesh/common/EventMeshThreadFactory.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    /**
+     * Constructs a new {@code Thread}.  Implementations may also initialize
+     * priority, name, daemon status, {@code ThreadGroup}, etc.
+     *
+     * @param runnable a runnable to be executed by new thread instance
+     * @return constructed thread, or {@code null} if the request to
+     * create a thread is rejected
+     */
+    @Override
+    public Thread newThread(Runnable runnable) {
```
Comment (jonyangx) [inline eventmesh-common/src/main/java/org/apache/eventmesh/common/EventMeshThreadFactory.java:67]: Suggest add abstract method newThread(final Runnable runnable ) in parent Class.

File: eventmesh-common/src/main/java/org/apache/eventmesh/common/ThreadPoolFactory.java [hunk 2]
```diff
@@ -24,9 +24,6 @@
 import java.util.concurrent.ThreadFactory;
 import java.util.concurrent.ThreadPoolExecutor;
 import java.util.concurrent.TimeUnit;
-import java.util.concurrent.atomic.AtomicInteger;
-
-import com.google.common.util.concurrent.ThreadFactoryBuilder;
 
 public class ThreadPoolFactory {
```
Comment (jonyangx) [inline eventmesh-common/src/main/java/org/apache/eventmesh/common/ThreadPoolFactory.java:28]: It's look like a utility Class.

File: eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/tcp/common/TcpClient.java [hunk 3]
```diff
@@ -75,8 +75,7 @@ public abstract class TcpClient implements Closeable {
     private transient ScheduledFuture<?> heartTask;
 
     protected static final ScheduledThreadPoolExecutor scheduler = new ScheduledThreadPoolExecutor(
```
Comment (jonyangx) [inline eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/tcp/common/TcpClient.java:77]: Suggest add createScheduledThreadPoolExecutor in Threadfactory Class.

Comment (mxsm) [reply eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/tcp/common/TcpClient.java:77]: I will fix those problems and resubmit this pr later

File: eventmesh-common/src/main/java/org/apache/eventmesh/common/EventMeshThreadFactory.java (hunk 1 above)
Comment (mxsm) [reply eventmesh-common/src/main/java/org/apache/eventmesh/common/EventMeshThreadFactory.java:67]: newThread method belongs to java.util.concurrent.ThreadFactory, and EventMeshThreadFactory is largely unchanged. so add `newThread(final Runnable runnable )` is not necessary.

**Comment by jonyangx** [comment]
@mxsm pls fix checkstyle check error.

**Comment by mxsm** [comment]
@jonyangx  done, the test case not pass, this pr has fix it #3058

**Review by jonyangx — APPROVED** [review]

**Pull request merged**